From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ricardo Pardini <ricardo@pardini.net>
Date: Sat, 2 Sep 2023 14:47:33 +0200
Subject: [ARCHEOLOGY] `uefi-arm64`/`edge`: bump to 6.5.y; rebase all patches
 vs `v6.5`(.0); fix Phytium's stmmac patch

> X-Git-Archeology: > recovered message: > - hammer patch `driver-phytium-stmmac-acpi.patch`
> X-Git-Archeology: > recovered message: > - use `.remove_new` due to "net: stmmac: Make stmmac_pltfr_remove() return void"
> X-Git-Archeology: - Revision f1b41bb771bf942c96a7b3838d56d8d5fadda469: https://github.com/armbian/build/commit/f1b41bb771bf942c96a7b3838d56d8d5fadda469
> X-Git-Archeology:   Date: Sat, 02 Sep 2023 14:47:33 +0200
> X-Git-Archeology:   From: Ricardo Pardini <ricardo@pardini.net>
> X-Git-Archeology:   Subject: `uefi-arm64`/`edge`: bump to 6.5.y; rebase all patches vs `v6.5`(.0); fix Phytium's stmmac patch
> X-Git-Archeology:
> X-Git-Archeology: - Revision a3c687e9a356050dc85fe509c6413548cb787c35: https://github.com/armbian/build/commit/a3c687e9a356050dc85fe509c6413548cb787c35
> X-Git-Archeology:   Date: Sat, 02 Sep 2023 14:48:25 +0200
> X-Git-Archeology:   From: Kevin Schmidt <kevin.patrick.schmidt@googlemail.com>
> X-Git-Archeology:   Subject: Add board: HiKey960
> X-Git-Archeology:
---
 arch/arm64/boot/dts/hisilicon/hi3660-hikey960.dts | 35 +++++++++-
 1 file changed, 33 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/hisilicon/hi3660-hikey960.dts b/arch/arm64/boot/dts/hisilicon/hi3660-hikey960.dts
index 111111111111..222222222222 100644
--- a/arch/arm64/boot/dts/hisilicon/hi3660-hikey960.dts
+++ b/arch/arm64/boot/dts/hisilicon/hi3660-hikey960.dts
@@ -197,6 +197,37 @@ optee {
 			method = "smc";
 		};
 	};
+
+	usb_hub_vdd: usb_hub_vdd {
+		compatible = "regulator-fixed";
+		regulator-name = "hub-vdd";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+		gpio = <&gpio5 6 0>;
+		enable-active-high;
+	};
+
+	usb-hub {
+		compatible = "hisilicon,usbhub";
+		typec-vbus-gpios = <&gpio25 2 GPIO_ACTIVE_HIGH>;
+		otg-switch-gpios = <&gpio25 6 GPIO_ACTIVE_HIGH>;
+		hub-vdd-supply = <&usb_hub_vdd>;
+		usb-role-switch;
+
+		port {
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			hikey_usb_ep0: endpoint@0 {
+				reg = <0>;
+				remote-endpoint = <&dwc3_role_switch>;
+			};
+			hikey_usb_ep1: endpoint@1 {
+				reg = <1>;
+				remote-endpoint = <&rt1711h_ep>;
+			};
+		};
+	};
 };
 
 /*
@@ -564,7 +595,7 @@ port {
 
 			rt1711h_ep: endpoint@0 {
 				reg = <0>;
-				remote-endpoint = <&dwc3_role_switch>;
+				remote-endpoint = <&hikey_usb_ep1>;
 			};
 		};
 	};
@@ -686,7 +717,7 @@ port {
 		#size-cells = <0>;
 		dwc3_role_switch: endpoint@0 {
 			reg = <0>;
-			remote-endpoint = <&rt1711h_ep>;
+			remote-endpoint = <&hikey_usb_ep0>;
 		};
 
 		dwc3_ss: endpoint@1 {
-- 
Armbian

